app/main/lib/text_similarity.py

```diff
@@ -16,7 +16,7 @@ def get_document_body(body):
     context = body.get("context", {})
     if context:
       body["contexts"] = [context]
-    if model_key != 'elasticsearch':
+    if model_key != 'opensearch' and model_key != 'elasticsearch':
```

Comment:
Unnecessary because `model_key` should never be `opensearch` this is an invalid input as far as the Alegre API is concerned..